### ISSUE TITLE
Use an inputObjectType as type for another inputObjectType

### ISSUE THREAD

**Issue by israelglar** [prisma/nexus#68]
If I try to use a inputTypeObjectType as a field for another inputObjectType like this:

```ts
export const ObjWithName = inputObjectType({
  name: 'ObjWithName',
  definition(t) {
    t.string('name');
  },
});

export const RootType = inputObjectType({
  name: 'RootType',
  definition(t) {
    t.field('obj', {type: ObjWithName});
  },
});
```

I get this error:
```ts
Error:(39, 21) TS2322: Type 'NexusInputObjectTypeDef<"ObjWithName">' is not assignable to type "Boolean" | "Float" | "ID" | "Int" | "JSON" | "String"'.
  Type 'NexusInputObjectTypeDef<"ObjWithName">' is not assignable to type '"String"'.
```

But I **don't** het this error if I try the same thing with `objectType`:
```ts
export const ObjWithName = objectType({
  name: 'ObjWithName',
  definition(t) {
    t.string('name');
  },
});

export const RootType = objectType({
  name: 'RootType',
  definition(t) {
    t.field('obj', {type: ObjWithName});
  },
});
```

I tried playing around with the type [here](https://github.com/prisma/nexus/blob/develop/src/definitions/definitionBlocks.ts#L242-L244) to look more like [this](https://github.com/prisma/nexus/blob/develop/src/definitions/definitionBlocks.ts#L67-L75). But I couldn't make it work to make a PR x.x

Thanks!

**Comment by tgriesser** [comment]
Cool, I agree this should work, will investigate!

**Comment by yakovlevyuri** [comment]
I'm facing the same issue. Thanks @tgriesser in advance!

**Comment by tgriesser** [comment]
Fixed & released in 0.11
